verifier/src/channel.rs

```diff
@@ -343,16 +342,59 @@ impl<E: FieldElement, H: ElementHasher<BaseField = E::BaseField>> ConstraintQuer
 // TRACE OUT-OF-DOMAIN FRAME
 // ================================================================================================
 
-struct TraceOodFrame<E: FieldElement> {
-    main_frame: EvaluationFrame<E>,
-    aux_frame: Option<EvaluationFrame<E>>,
+pub struct TraceOodFrame<E: FieldElement> {
+    values: Vec<E>,
+    main_trace_width: usize,
+    aux_trace_width: usize,
 }
 
 impl<E: FieldElement> TraceOodFrame<E> {
-    pub fn new(main_frame: EvaluationFrame<E>, aux_frame: Option<EvaluationFrame<E>>) -> Self {
+    pub fn new(values: Vec<E>, main_trace_width: usize, aux_trace_width: usize) -> Self {
         Self {
-            main_frame,
-            aux_frame,
+            values,
+            main_trace_width,
+            aux_trace_width,
+        }
+    }
+
+    pub fn values(&self) -> &[E] {
+        &self.values
+    }
+
+    pub fn main_frame(&self) -> EvaluationFrame<E> {
```

Comment:
minor nit: maybe add a brief comment explaining why we do it this way (and maybe mentioning `OodFrame` so that people can easily find where the "other side" of this code is)?

Reply:
Let me know if the comments makes sense or if they are too much verbose.

Reply:
This looks awesome! Thank you!